[2025-06-13T17:05:49.219+0200] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: agile_data_science_batch_prediction_model_training.pyspark_train_classifier_model manual__2025-06-11T16:07:32.404313+00:00 [queued]>
[2025-06-13T17:05:49.240+0200] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: agile_data_science_batch_prediction_model_training.pyspark_train_classifier_model manual__2025-06-11T16:07:32.404313+00:00 [queued]>
[2025-06-13T17:05:49.240+0200] {taskinstance.py:1279} INFO - 
--------------------------------------------------------------------------------
[2025-06-13T17:05:49.240+0200] {taskinstance.py:1280} INFO - Starting attempt 3 of 4
[2025-06-13T17:05:49.240+0200] {taskinstance.py:1281} INFO - 
--------------------------------------------------------------------------------
[2025-06-13T17:05:49.265+0200] {taskinstance.py:1300} INFO - Executing <Task(BashOperator): pyspark_train_classifier_model> on 2025-06-11 16:07:32.404313+00:00
[2025-06-13T17:05:49.268+0200] {standard_task_runner.py:55} INFO - Started process 29294 to run task
[2025-06-13T17:05:49.275+0200] {standard_task_runner.py:82} INFO - Running: ['airflow', 'tasks', 'run', 'agile_data_science_batch_prediction_model_training', 'pyspark_train_classifier_model', 'manual__2025-06-11T16:07:32.404313+00:00', '--job-id', '49', '--raw', '--subdir', 'DAGS_FOLDER/setup.py', '--cfg-path', '/tmp/tmprauri4m4']
[2025-06-13T17:05:49.276+0200] {standard_task_runner.py:83} INFO - Job 49: Subtask pyspark_train_classifier_model
[2025-06-13T17:05:49.366+0200] {task_command.py:388} INFO - Running <TaskInstance: agile_data_science_batch_prediction_model_training.pyspark_train_classifier_model manual__2025-06-11T16:07:32.404313+00:00 [running]> on host l089.lab.dit.upm.es
[2025-06-13T17:05:49.485+0200] {taskinstance.py:1507} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=airflow
AIRFLOW_CTX_DAG_ID=agile_data_science_batch_prediction_model_training
AIRFLOW_CTX_TASK_ID=pyspark_train_classifier_model
AIRFLOW_CTX_EXECUTION_DATE=2025-06-11T16:07:32.404313+00:00
AIRFLOW_CTX_TRY_NUMBER=3
AIRFLOW_CTX_DAG_RUN_ID=manual__2025-06-11T16:07:32.404313+00:00
[2025-06-13T17:05:49.487+0200] {subprocess.py:63} INFO - Tmp dir root location: 
 /tmp
[2025-06-13T17:05:49.487+0200] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', '\nspark-submit --master local[4]   /home/monica.fernandez/practica_creativa//resources/train_spark_mllib_model.py   /home/monica.fernandez/practica_creativa/']
[2025-06-13T17:05:49.496+0200] {subprocess.py:86} INFO - Output:
[2025-06-13T17:05:52.394+0200] {subprocess.py:93} INFO - 25/06/13 17:05:52 WARN Utils: Your hostname, l089 resolves to a loopback address: 127.0.1.1; using 138.4.31.89 instead (on interface enp3s0)
[2025-06-13T17:05:52.398+0200] {subprocess.py:93} INFO - 25/06/13 17:05:52 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address
[2025-06-13T17:05:56.628+0200] {subprocess.py:93} INFO - 25/06/13 17:05:56 INFO SparkContext: Running Spark version 3.5.3
[2025-06-13T17:05:56.629+0200] {subprocess.py:93} INFO - 25/06/13 17:05:56 INFO SparkContext: OS info Linux, 6.1.0-32-amd64, amd64
[2025-06-13T17:05:56.629+0200] {subprocess.py:93} INFO - 25/06/13 17:05:56 INFO SparkContext: Java version 17.0.14
[2025-06-13T17:05:56.735+0200] {subprocess.py:93} INFO - 25/06/13 17:05:56 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-06-13T17:05:56.923+0200] {subprocess.py:93} INFO - 25/06/13 17:05:56 INFO ResourceUtils: ==============================================================
[2025-06-13T17:05:56.923+0200] {subprocess.py:93} INFO - 25/06/13 17:05:56 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-06-13T17:05:56.924+0200] {subprocess.py:93} INFO - 25/06/13 17:05:56 INFO ResourceUtils: ==============================================================
[2025-06-13T17:05:56.925+0200] {subprocess.py:93} INFO - 25/06/13 17:05:56 INFO SparkContext: Submitted application: train_spark_mllib_model.py
[2025-06-13T17:05:56.968+0200] {subprocess.py:93} INFO - 25/06/13 17:05:56 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-06-13T17:05:56.978+0200] {subprocess.py:93} INFO - 25/06/13 17:05:56 INFO ResourceProfile: Limiting resource is cpu
[2025-06-13T17:05:56.986+0200] {subprocess.py:93} INFO - 25/06/13 17:05:56 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-06-13T17:05:57.064+0200] {subprocess.py:93} INFO - 25/06/13 17:05:57 INFO SecurityManager: Changing view acls to: monica.fernandez
[2025-06-13T17:05:57.066+0200] {subprocess.py:93} INFO - 25/06/13 17:05:57 INFO SecurityManager: Changing modify acls to: monica.fernandez
[2025-06-13T17:05:57.067+0200] {subprocess.py:93} INFO - 25/06/13 17:05:57 INFO SecurityManager: Changing view acls groups to:
[2025-06-13T17:05:57.068+0200] {subprocess.py:93} INFO - 25/06/13 17:05:57 INFO SecurityManager: Changing modify acls groups to:
[2025-06-13T17:05:57.068+0200] {subprocess.py:93} INFO - 25/06/13 17:05:57 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: monica.fernandez; groups with view permissions: EMPTY; users with modify permissions: monica.fernandez; groups with modify permissions: EMPTY
[2025-06-13T17:05:57.567+0200] {subprocess.py:93} INFO - 25/06/13 17:05:57 INFO Utils: Successfully started service 'sparkDriver' on port 39517.
[2025-06-13T17:05:57.684+0200] {subprocess.py:93} INFO - 25/06/13 17:05:57 INFO SparkEnv: Registering MapOutputTracker
[2025-06-13T17:05:57.808+0200] {subprocess.py:93} INFO - 25/06/13 17:05:57 INFO SparkEnv: Registering BlockManagerMaster
[2025-06-13T17:05:57.859+0200] {subprocess.py:93} INFO - 25/06/13 17:05:57 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-06-13T17:05:57.861+0200] {subprocess.py:93} INFO - 25/06/13 17:05:57 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-06-13T17:05:57.875+0200] {subprocess.py:93} INFO - 25/06/13 17:05:57 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-06-13T17:05:57.943+0200] {subprocess.py:93} INFO - 25/06/13 17:05:57 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-0dad0bfa-5f59-449c-9cb1-c1014f2afb79
[2025-06-13T17:05:57.978+0200] {subprocess.py:93} INFO - 25/06/13 17:05:57 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-06-13T17:05:58.033+0200] {subprocess.py:93} INFO - 25/06/13 17:05:58 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-06-13T17:05:58.295+0200] {subprocess.py:93} INFO - 25/06/13 17:05:58 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2025-06-13T17:05:58.422+0200] {subprocess.py:93} INFO - 25/06/13 17:05:58 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
[2025-06-13T17:05:58.437+0200] {subprocess.py:93} INFO - 25/06/13 17:05:58 INFO Utils: Successfully started service 'SparkUI' on port 4041.
[2025-06-13T17:05:58.634+0200] {subprocess.py:93} INFO - 25/06/13 17:05:58 INFO Executor: Starting executor ID driver on host 138.4.31.89
[2025-06-13T17:05:58.634+0200] {subprocess.py:93} INFO - 25/06/13 17:05:58 INFO Executor: OS info Linux, 6.1.0-32-amd64, amd64
[2025-06-13T17:05:58.635+0200] {subprocess.py:93} INFO - 25/06/13 17:05:58 INFO Executor: Java version 17.0.14
[2025-06-13T17:05:58.645+0200] {subprocess.py:93} INFO - 25/06/13 17:05:58 INFO Executor: Starting executor with user classpath (userClassPathFirst = false): ''
[2025-06-13T17:05:58.646+0200] {subprocess.py:93} INFO - 25/06/13 17:05:58 INFO Executor: Created or updated repl class loader org.apache.spark.util.MutableURLClassLoader@1a76123 for default.
[2025-06-13T17:05:58.686+0200] {subprocess.py:93} INFO - 25/06/13 17:05:58 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 33141.
[2025-06-13T17:05:58.687+0200] {subprocess.py:93} INFO - 25/06/13 17:05:58 INFO NettyBlockTransferService: Server created on 138.4.31.89:33141
[2025-06-13T17:05:58.689+0200] {subprocess.py:93} INFO - 25/06/13 17:05:58 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-06-13T17:05:58.699+0200] {subprocess.py:93} INFO - 25/06/13 17:05:58 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 138.4.31.89, 33141, None)
[2025-06-13T17:05:58.709+0200] {subprocess.py:93} INFO - 25/06/13 17:05:58 INFO BlockManagerMasterEndpoint: Registering block manager 138.4.31.89:33141 with 434.4 MiB RAM, BlockManagerId(driver, 138.4.31.89, 33141, None)
[2025-06-13T17:05:58.712+0200] {subprocess.py:93} INFO - 25/06/13 17:05:58 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 138.4.31.89, 33141, None)
[2025-06-13T17:05:58.713+0200] {subprocess.py:93} INFO - 25/06/13 17:05:58 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 138.4.31.89, 33141, None)
[2025-06-13T17:05:59.799+0200] {subprocess.py:93} INFO - 25/06/13 17:05:59 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2025-06-13T17:05:59.811+0200] {subprocess.py:93} INFO - 25/06/13 17:05:59 INFO SharedState: Warehouse path is 'file:/tmp/airflowtmppz2nuy2z/spark-warehouse'.
[2025-06-13T17:06:01.906+0200] {subprocess.py:93} INFO - MLflow Run ID: 73a7149e0a2642ea8d7141a72e2024b1, Experiment ID: 0
[2025-06-13T17:06:01.906+0200] {subprocess.py:93} INFO - MLflow Tracking URI: file:///tmp/airflowtmppz2nuy2z/mlruns
[2025-06-13T17:06:02.992+0200] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-06-13T17:06:02.993+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/practica_creativa/resources/train_spark_mllib_model.py", line 216, in <module>
[2025-06-13T17:06:02.993+0200] {subprocess.py:93} INFO -     main(sys.argv[1])
[2025-06-13T17:06:02.993+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/practica_creativa/resources/train_spark_mllib_model.py", line 65, in main
[2025-06-13T17:06:02.993+0200] {subprocess.py:93} INFO -     features = spark.read.json(input_path, schema=schema)
[2025-06-13T17:06:02.994+0200] {subprocess.py:93} INFO -                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-06-13T17:06:02.994+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/.sdkman/candidates/spark/current/python/lib/pyspark.zip/pyspark/sql/readwriter.py", line 425, in json
[2025-06-13T17:06:02.994+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/.sdkman/candidates/spark/current/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2025-06-13T17:06:02.994+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/.sdkman/candidates/spark/current/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 185, in deco
[2025-06-13T17:06:03.011+0200] {subprocess.py:93} INFO - pyspark.errors.exceptions.captured.AnalysisException: [PATH_NOT_FOUND] Path does not exist: file:/data/simple_flight_delay_features.jsonl.bz2.
[2025-06-13T17:06:03.416+0200] {subprocess.py:93} INFO - 25/06/13 17:06:03 INFO SparkContext: Invoking stop() from shutdown hook
[2025-06-13T17:06:03.432+0200] {subprocess.py:93} INFO - 25/06/13 17:06:03 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2025-06-13T17:06:03.466+0200] {subprocess.py:93} INFO - 25/06/13 17:06:03 INFO SparkUI: Stopped Spark web UI at http://138.4.31.89:4041
[2025-06-13T17:06:03.509+0200] {subprocess.py:93} INFO - 25/06/13 17:06:03 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2025-06-13T17:06:03.589+0200] {subprocess.py:93} INFO - 25/06/13 17:06:03 INFO MemoryStore: MemoryStore cleared
[2025-06-13T17:06:03.590+0200] {subprocess.py:93} INFO - 25/06/13 17:06:03 INFO BlockManager: BlockManager stopped
[2025-06-13T17:06:03.599+0200] {subprocess.py:93} INFO - 25/06/13 17:06:03 INFO BlockManagerMaster: BlockManagerMaster stopped
[2025-06-13T17:06:03.603+0200] {subprocess.py:93} INFO - 25/06/13 17:06:03 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2025-06-13T17:06:03.708+0200] {subprocess.py:93} INFO - 25/06/13 17:06:03 INFO SparkContext: Successfully stopped SparkContext
[2025-06-13T17:06:03.709+0200] {subprocess.py:93} INFO - 25/06/13 17:06:03 INFO ShutdownHookManager: Shutdown hook called
[2025-06-13T17:06:03.710+0200] {subprocess.py:93} INFO - 25/06/13 17:06:03 INFO ShutdownHookManager: Deleting directory /tmp/spark-aba4c9f0-6b1d-4574-88e0-4d25c9e19fdf
[2025-06-13T17:06:03.719+0200] {subprocess.py:93} INFO - 25/06/13 17:06:03 INFO ShutdownHookManager: Deleting directory /tmp/spark-aba4c9f0-6b1d-4574-88e0-4d25c9e19fdf/pyspark-e4f2883e-0fe5-43ec-ac2c-71249aa6b2f9
[2025-06-13T17:06:03.725+0200] {subprocess.py:93} INFO - 25/06/13 17:06:03 INFO ShutdownHookManager: Deleting directory /tmp/spark-55cb46cb-f368-4c73-a6cb-b2b0f88a9b46
[2025-06-13T17:06:03.786+0200] {subprocess.py:97} INFO - Command exited with return code 1
[2025-06-13T17:06:03.810+0200] {taskinstance.py:1768} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/monica.fernandez/practica_creativa/venv-airflow/lib/python3.11/site-packages/airflow/operators/bash.py", line 196, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-06-13T17:06:03.818+0200] {taskinstance.py:1318} INFO - Marking task as UP_FOR_RETRY. dag_id=agile_data_science_batch_prediction_model_training, task_id=pyspark_train_classifier_model, execution_date=20250611T160732, start_date=20250613T150549, end_date=20250613T150603
[2025-06-13T17:06:03.884+0200] {standard_task_runner.py:100} ERROR - Failed to execute job 49 for task pyspark_train_classifier_model (Bash command failed. The command returned a non-zero exit code 1.; 29294)
[2025-06-13T17:06:03.957+0200] {local_task_job.py:208} INFO - Task exited with return code 1
[2025-06-13T17:06:04.008+0200] {taskinstance.py:2578} INFO - 0 downstream tasks scheduled from follow-on schedule check
